packages/nimble-components/src/table-column/icon/tests/table-column-icon.spec.ts

```diff
@@ -0,0 +1,443 @@
+                        element
+                    );
+                    await element.setData([{ field1: 'a' }]);
+                    await connect();
+                    await waitForUpdatesAsync();
+                    (element.columns[0] as TableColumnIcon).groupIndex = 0;
+                    await waitForUpdatesAsync();
+
+                    expect(
+                        pageObject.getRenderedGroupHeaderContent(0)
+                    ).toContain(value.name);
+                }
+            );
+        }
+    });
+
+    it('sets group header text to key value when unmatched (instead of default)', async () => {
+        ({ element, connect, disconnect } = await setup([
+            { defaultMapping: true, label: 'bravo', icon: 'nimble-icon-xmark' }
+        ]));
+        pageObject = new TablePageObject<SimpleTableRecord>(element);
+        await element.setData([{ field1: 'unmatched' }]);
+        await connect();
+        await waitForUpdatesAsync();
+        (element.columns[0] as TableColumnIcon).groupIndex = 0;
+        await waitForUpdatesAsync();
+
+        expect(pageObject.getRenderedGroupHeaderContent(0)).toContain(
+            'unmatched'
+        );
+    });
+
+    describe('validation', () => {
+        it('is valid with no mappings', async () => {
+            ({ element, connect, disconnect } = await setup([], 'number'));
+            await connect();
+            await waitForUpdatesAsync();
+            const column = element.columns[0] as TableColumnIcon;
+            expect(column.checkValidity()).toBeTrue();
+            expect(column.validity.invalidMappingKeyValueForType).toBeFalse();
+            expect(column.validity.multipleDefaultMappings).toBeFalse();
+            expect(column.validity.unsupportedMappingType).toBeFalse();
+            expect(column.validity.duplicateMappingKey).toBeFalse();
+            expect(column.validity.missingKeyValue).toBeFalse();
+            expect(column.validity.invalidIconName).toBeFalse();
+        });
+
+        it('is valid with valid numeric key values', async () => {
+            ({ element, connect, disconnect } = await setup(
+                [
+                    { key: '0', label: 'alpha', icon: 'nimble-icon-xmark' },
+                    { key: '1', label: 'alpha', icon: 'nimble-icon-xmark' },
+                    { key: '1.01', label: 'alpha', icon: 'nimble-icon-xmark' },
+                    { key: '-1.01', label: 'alpha', icon: 'nimble-icon-xmark' },
+                    { key: '-1e3', label: 'alpha', icon: 'nimble-icon-xmark' }
+                ],
+                'number'
+            ));
+            await connect();
+            await waitForUpdatesAsync();
+            const column = element.columns[0] as TableColumnIcon;
+            expect(column.checkValidity()).toBeTrue();
+            expect(column.validity.invalidMappingKeyValueForType).toBeFalse();
+        });
+
+        it('catches invalid numeric key values', async () => {
```

Comment:
nit: the icon column only tests for invalid numeric values but the enum column tests for both invalid numerics and invalid booleans. I'd expect either:
- one column handles all the testing of the base class logic and the other one does no testing of it
- both columns have equivalent tests

I'm fine with either but right now it seems like a slightly inconsistent mix.

Reply:
Since they both have separate ColumnValidator types, I guess it makes sense to test both. I've added the invalid booleans tests to the other one.